### ISSUE TITLE
Prefix dropdown in checkout doesn't select empty value by default if required

### ISSUE THREAD

**Issue by DanieliMi** [magento/magento2#18823]
### Preconditions (*)

1. Magento 2.2.6
2. Luma Theme

### Steps to reproduce (*)

1. In Admin Backend go to Stores > Configuration > Customers > Name and Address Options
2. Set "Show Prefix" to "Required"
3. Set "Prefix Dropdown Options" to ";Mr.;Mrs."
4. Save Config
5. Go to your store
6. Add a item to your cart
7. Go to checkout

### Expected result (*)

I expect the empty value in the prefix dropdown to be selected by default.
![prefix_expected](https://user-images.githubusercontent.com/31039652/47552616-69545b00-d905-11e8-83ff-4f6d65bae755.png)

### Actual result (*)

Instead of having the empty value selected by default it will be always the first value selected.
![prefix_actual](https://user-images.githubusercontent.com/31039652/47552698-9e60ad80-d905-11e8-8d97-afb9434496e1.png)

### Additional information
![issue_18823](https://user-images.githubusercontent.com/3409584/83544354-5f241680-a4c3-11ea-9bb7-15f3797a4259.png)

**Comment by magento-engcom-team** [comment]
Hi @DanieliMi. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@DanieliMi do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by DanieliMi** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @DanieliMi. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @DanieliMi, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18823-2-2-6/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18823-2-2-6//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18823-2-2-6//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by DanieliMi** [comment]
Yes i can confirm i was able to reproduce the problem.

**Comment by magento-engcom-team** [comment]
Hi @hardik-krish. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by hardik-krish** [comment]
Yes i am also able to reproduce this issue. Same things working fine in customer registration area.

**Comment by magento-engcom-team** [comment]
@hardik-krish Thank you for verifying the issue.

Unfortunately, not enough information was provided to created internal ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess
- [x] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by hardik-krish** [comment]
I have already resolved this issue check this #18885 but that didn't reviewed yet. it was above one month ago i have resolved this issue.

**Comment by ghost** [comment]
@hardik-krish  i'm verify this issue.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97012`, `MAGETWO-97013` were created

**Comment by hardik-krish** [comment]
#18885 I have resolved this ticket.before you confirm so please review it and let me know if you found something wrong.

**Comment by diazwatson** [comment]
@sivaschenko 

> I think that the fix for the issue should be specific to prefix field, correct data with appropriate placeholder should be used for a field.

Is this what you mean?
https://github.com/magento/magento2/blob/18dc53fa7366aa0890a497ec42c8cf872c79cdab/app/code/Magento/Customer/Model/Options.php#L107
```
        if ($isOptional && trim(current($options))) {
            $result = array_merge([' ' => __("-- Please Select --")->getText()], $result);
        }
```
Although and as shown in the code this option won't be added when the field not isOptional

**Comment by sivaschenko** [comment]
@diazwatson yes, that's what I meant. However, the code you are referencing is not based on options specified in the configuration, it is adding an additional option for an optional field to enable user not to select any options.

To fix this issue we probably will need to change the filtering of the provided option values.

**Comment by ajithkumar-maragathavel** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ajithkumar-maragathavel. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ajithkumar-maragathavel, here is your [Magento instance](https://i-18823-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-18823-2-4-develop.instances.magento-community.engineering/admin_48bb](https://i-18823-2-4-develop.instances.magento-community.engineering/admin_48bb)
Login: `4a22aa9f` Password: `248f137f6031`
Instance will be terminated in up to 3 hours.

**Comment by ajithkumar-maragathavel** [comment]
@magento-admin Please assign this ticket to me. I will work

**Comment by sdzhepa** [comment]
![issue_18823](https://user-images.githubusercontent.com/3409584/83544354-5f241680-a4c3-11ea-9bb7-15f3797a4259.png)

**Comment by magento-engcom-team** [comment]
Hi @DanieliMi. Thank you for your report. 
The issue has been fixed in magento/magento2#28238 by @engcom-Charlie in 2.4-develop branch
Related commit(s):
 - [7aa9a9c35990d4f7115661a03a1cce39733a21c9](https://github.com/magento/magento2/commit/7aa9a9c35990d4f7115661a03a1cce39733a21c9)
 - [daecbb410a7481d8ea0e80e8c57479e02beb93b4](https://github.com/magento/magento2/commit/daecbb410a7481d8ea0e80e8c57479e02beb93b4)
 - [f208eaefe695fbd969f669704a80c2ec0f75c3b2](https://github.com/magento/magento2/commit/f208eaefe695fbd969f669704a80c2ec0f75c3b2)
 - [e5be53f628f6170324572132bf66742b6fb63115](https://github.com/magento/magento2/commit/e5be53f628f6170324572132bf66742b6fb63115)
 - [0213d58d384c2bf7d780f73b057c9b0096710db0](https://github.com/magento/magento2/commit/0213d58d384c2bf7d780f73b057c9b0096710db0)
 - [e322c586a77db8ccf2ec35048cd9bfa48747469f](https://github.com/magento/magento2/commit/e322c586a77db8ccf2ec35048cd9bfa48747469f)

The fix will be available with the upcoming 2.4.1 release.
